packages/core/src/shared/lsp/manifestResolver.ts

```diff
@@ -90,25 +99,47 @@ export class ManifestResolver {
             return JSON.parse(content) as Manifest
         } catch (error) {
             throw new ToolkitError(
-                `Failed to parse manifest: ${error instanceof Error ? error.message : 'Unknown error'}`
+                `Failed to parse ${this.lsName} manifest: ${error instanceof Error ? error.message : 'Unknown error'}`
             )
         }
     }
 
+    /**
+     * Check if the current manifest is deprecated.
+     * If yes and user hasn't muted this notification, shows a toast message with two buttons:
+     * - OK: close and do nothing
+     * - Don't Show Again: Update global state (muteDecprecation) so the deprecation message is never shown for this manifest.
+     * @param manifest
+     */
     private checkDeprecation(manifest: Manifest): void {
-        if (manifest.isManifestDeprecated) {
-            logger.info('This LSP manifest is deprecated. No future updates will be available.')
+        if (manifest.isManifestDeprecated && !this.getStorage()[this.lsName].muteDeprecation) {
+            const deprecationMessage = `${this.lsName} manifest is deprecated. No future updates will be available.`
+            logger.info(deprecationMessage)
```

Comment:
We might want to log the deprecated message regardless of whether or not they muted it just to give them some sort of heads up if they look at the logs. Otherwise people are just going to press don't show again and then never remember